### PR TITLE
network: several follow-ups for recently merged PRs

### PR DESCRIPTION
**Pull request by yuwata**
Follow-ups for #18131 and #17635.

### CODE REVIEW

**Review by bluca — APPROVED** [review]
good catches, especially the pointer one, missed it - lgtm

File: src/network/networkd-route.c [hunk 1]
```diff
@@ -2121,6 +2121,7 @@ int config_parse_tcp_advmss(
 
         if (isempty(rvalue)) {
                 n->advmss = 0;
+                TAKE_PTR(n);
```
Comment (ssahani) [inline src/network/networkd-route.c:2130]: No need to store zero values as it has no use.

Comment (yuwata) [reply src/network/networkd-route.c:2130]: You mean, we should drop `if (isempty(rvalue))` block? That's also a good way. If you like it, I will drop the block.

Comment (ssahani) [reply src/network/networkd-route.c:2130]: What you think ? Should we go ahead with rest values which  good or drop it altogether. Hmm pretty confusing

Comment (yuwata) [reply src/network/networkd-route.c:2130]: Hm? The problem here is, if we do not have `TAKE_PTR()` in this block, then setting advmss with an empty string causes the [Route] section invalid. So, we need to do one of the following:
1. add `TAKE_PTR()` in the block, then we accept an empty string for the setting.
2. drop the `if` block to refuse an empty string.

I have no preference. So, if you like the second way, I will drop the block.

Comment (ssahani) [reply src/network/networkd-route.c:2130]: Ok lets allow this and for future too.

Comment (yuwata) [reply src/network/networkd-route.c:2130]: Thanks.

**Comment by yuwata** [comment]
Setting the green label, as @bluca approved this.

**Comment by poettering** [comment]
lgtm

**Pull request merged**